xds/internal/resolver/xds_resolver_test.go

```diff
@@ -213,11 +260,18 @@ func testSetup(t *testing.T, opts setupOpts) (*xdsResolver, *testClientConn, fun
 	}
 
 	tcc := newTestClientConn()
-	r, err := builder.Build(target, tcc, resolver.BuildOptions{})
+	tgt := target
+	if opts.target != nil {
+		tgt = *opts.target
+	}
```

Comment:
This is a bit of magic where we get the `target` from a global var if not set in `opts`. This makes the tests harder to read since the reader will have to keep track of all this. Instead, could we just always pass a valid `target` from tests?

Reply:
Done